beacon-chain/sync/backfill/blobs.go

```diff
@@ -0,0 +1,134 @@
+package backfill
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/das"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/db/filesystem"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/verification"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/encoding/bytesutil"
+)
+
+var (
+	errUnexpectedResponseSize    = errors.New("received more blobs than expected for the requested range")
+	errUnexpectedCommitment      = errors.New("BlobSidecar commitment does not match block")
+	errUnexpectedResponseContent = errors.New("BlobSidecar response does not include expected values in expected order")
+	errBatchVerifierMismatch     = errors.New("the list of blocks passed to the availability check does not match what was verified")
+)
+
+type blobSummary struct {
+	blockRoot  [32]byte
+	index      uint64
+	commitment [48]byte
+	signature  [fieldparams.BLSSignatureLength]byte
+}
+
+func newBlobSync(cs, retentionStart primitives.Slot, vbs verifiedROBlocks, nbv verification.NewBlobVerifier, st *filesystem.BlobStorage) (*blobSync, error) {
+	todo, err := vbs.blobIdents(retentionStart)
+	if err != nil {
+		return nil, err
+	}
+	bbv := newBlobBatchVerifier(nbv)
+	as := das.NewLazilyPersistentStore(st, bbv)
+	return &blobSync{cs: cs, expected: todo, bbv: bbv, store: as}, nil
+}
+
+type blobVerifierMap map[[32]byte][fieldparams.MaxBlobsPerBlock]verification.BlobVerifier
+
+type blobSync struct {
+	store    das.AvailabilityStore
+	expected []blobSummary
+	next     int
+	bbv      *blobBatchVerifier
+	cs       primitives.Slot
+}
+
+func (bs *blobSync) blobsNeeded() int {
+	return len(bs.expected) - bs.next
+}
+
+func (bs *blobSync) validateNext(rb blocks.ROBlob) error {
```

Comment:
Where do we verify the blob's slot?

Reply:
We implicitly know that they match by comparing the block roots. The base request method also adds the following validations:
- https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/sync/rpc_send_request.go#L296-L304
- https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/sync/rpc_send_request.go#L221-L265